lte/gateway/python/magma/pipelined/openflow/messages.py

```diff
@@ -272,8 +274,19 @@ def _flow_matches_flowmsg(self, dp, flow, msg):
                            if type(i) == dp.ofproto_parser.OFPActionOutput]
             outputs_match = sorted(outputs_flow) == sorted(outputs_msg)
 
-        flow_match = all([flow.match.get(i, None) == msg.match.get(i, None)
-                          for i in MATCH_ATTRIBUTES])
+        match_flow = {key: flow.match.get(key) for key in MATCH_ATTRIBUTES
+                      if key in flow.match}
+        match_msg = {key: flow.match.get(key, None) for key in MATCH_ATTRIBUTES
+                     if key in msg.match}
+        # Filter out things that break equality, TODO move this to helper func
+        for key in list(match_flow.keys()):
+            if match_flow[key] == ('0.0.0.0', '0.0.0.0'):
+                del match_flow[key]
+        for key in list(match_msg.keys()):
+            if match_msg[key] == ('0.0.0.0', '0.0.0.0'):
+                del match_msg[key]
+
+        flow_match = match_flow == match_msg
         return flow_match and reg_loads_match and resubmits_match and \
```

Comment:
It would be better if we convert flow and msg to common format and then compare it.
for example OFPFlowStats has 'match' and 'instructions', flow object also has these members. we can convert it to json and compare it. so pipelineD does not need to special case certain flow members and the compare operation becomes generic.

Reply:
Not quite sure if I follow, this attempts to extract the common attributes and compares them.
How does converting to json solve it, we would still have the case where OFPFlowStats.match has `('0.0.0.0', '0.0.0.0')` value and flowmessage.match has None value, which for OVS are the same but would break the == comparison

Reply:
I see, I will think about it bit further.
Lets get this merged to fix it.